src/codegen/doc/format.rs

```diff
@@ -0,0 +1,73 @@
+use regex::{Captures, Regex};
+
+const LANGUAGE_SEP_END : &'static str = "\" -->";
+const LANGUAGE_BLOCK_BEGIN : &'static str = "|[<!-- language=\"";
+const LANGUAGE_BLOCK_END : &'static str = "\n]|";
+
+pub fn reformat_doc(input: &str) -> String {
+    let out = code_blocks_transformation(input);
+    replace_c_types(&out)
+}
+
+fn code_blocks_transformation(input: &str) -> String {
+    let mut out = String::new();
+
+    for (num, entry) in input.split(LANGUAGE_BLOCK_BEGIN).into_iter().enumerate() {
+        if num & 1 == 0 {
+            out.push_str(&replace_c_types(entry));
+            continue;
+        }
+        out.push_str("```");
+        for (num, entry) in get_language(entry, &mut out).split(LANGUAGE_BLOCK_END)
+                                                         .into_iter()
+                                                         .enumerate() {
+            if num & 1 == 0 {
+                out.push_str(entry);
+                continue;
+            }
+            out.push_str(&format!("\n```{}", replace_c_types(entry)));
+            break;
+        }
+    }
+    out
+}
+
+fn get_language<'a>(entry: &'a str, out: &mut String) -> &'a str {
+    if let Some(pos) = entry.find(LANGUAGE_SEP_END) {
+        out.push_str(&entry[0..pos]);
+        &entry[(pos + LANGUAGE_SEP_END.len())..]
+    } else {
+        out.push_str("\n");
+        entry
+    }
+}
+
+fn replace_c_types(entry: &str) -> String {
+    if let (Ok(reg), Ok(reg2)) = (Regex::new(r"(?:#)?(?:G[A-Z][a-z]|Gdk|Gtk)[\w]*"), Regex::new(r"@\w*(\s|\n)")) {
+        let out = &reg.replace_all(entry, |caps: &Captures| {
+            match caps.at(0) {
+                Some(mut s) => {
+                    if s.starts_with("#") {
```

Comment:
That's why I suggested `(?:G|Gdk|Gtk)([A-Z]\w*)`. Here the prefix must be followed by a capital. But it will be more reliable like this
```
(?:#|\b)(?:G|Gdk|Gtk)([A-Z]\w*)
```
No matching in the middle of a word now.